packaging_automation/update_package_properties.py

```diff
@@ -226,43 +271,69 @@ def update_all_changes(github_token: non_empty(non_blank(str)), package_properti
                        packaging_path: str):
     validate_package_properties_params_for_update_all_changes(package_properties_params)
     templates_path = f"{BASE_PATH}/templates"
-    update_pkgvars(package_properties_params.project_name, package_properties_params.project_version,
-                   package_properties_params.fancy, package_properties_params.fancy_version_number, templates_path,
+    update_pkgvars(package_properties_params, templates_path,
                    f"{packaging_path}")
-    latest_changelog = changelog_for_tag(github_token, package_properties_params.project_name, tag_name)
+    latest_changelog = changelog_for_tag(github_token,
+                                         package_properties_params.supported_project.value.github_repo_name, tag_name)
     package_properties_params.latest_changelog = latest_changelog
-
-    prepend_latest_changelog_into_debian_changelog(package_properties_params, f"{packaging_path}/debian/changelog")
+    if package_properties_params.changelog_add():
+        prepend_latest_changelog_into_debian_changelog(package_properties_params, f"{packaging_path}/debian/changelog")
     spec_full_path = f"{packaging_path}/{package_properties_params.spec_file_name()}"
     update_rpm_spec(package_properties_params, spec_full_path, templates_path)
 
 
-# update_all_changes("93647419346e194ae3094598139d68ebf2263ee0", "citus", "10.0.3", "v10.0.3", True, 1,
-#                    "gindibay@microsoft.com", "Gurkan Indibay")
+CHECKOUT_DIR = "update_properties_temp"
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
-    parser.add_argument('--gh_token')
-    parser.add_argument('--prj_name')
-    parser.add_argument('--tag_name')
-    parser.add_argument('--fancy')
-    parser.add_argument('--fancy_ver_no')
-    parser.add_argument('--email')
-    parser.add_argument('--name')
-    parser.add_argument('--date', nargs='?', default=datetime.now())
+    parser.add_argument('--gh_token', required=True)
+    parser.add_argument('--prj_name', choices=[r.name for r in SupportedProjects])
+    parser.add_argument('--tag_name', required=True)
+    parser.add_argument('--fancy_ver_no', type=int, choices=range(1, 10), default=1)
+    parser.add_argument('--email', required=True)
+    parser.add_argument('--name', required=True)
+    parser.add_argument('--date')
+    parser.add_argument("--pipeline", action="store_true")
```

Comment:
1) What is the usage of `pipeline` param and can we document it in README ?
2) I think we should remove `fancy` from README too

Reply:
we can execute this script either from our computer or in pipeline. When using pipeline parameter code is not cloned since pipeline clones it beforehand.

Reply:
I'll update Readme for all requests

Reply:
>I'll update Readme for all requests

waiting for the update then

Reply:
> we can execute this script either from our computer or in pipeline. When using pipeline parameter code is not cloned since pipeline clones it beforehand.

Is there any reason to not clone the repo always via the script, even when using pipeline on CI.
I think it might be a slightly cleaner that way, what do you think ?

Reply:
The problem is that pipeline will execute in the packaging repo so code is already there. If we checkout the code here again, we will do an unnecessary job. For example when I execute the pipeline if pipeline is true, exec path will be used which will be ${pwd} most probably.

Reply:
oh now understood what you mean, resolving the comment then